src/librustc_error_codes/error_codes/E0763.md

```diff
@@ -0,0 +1,32 @@
+Mutable references (`&mut`) can only be used in constant functions, not statics
+or constants.
```

Comment:
```suggestion
or constants. This limitation exists to prevent the creation of constants that 
have a mutable reference in their final value. If you had a constant of `&mut i32`
type, you could modify the value through that reference, making the constant
essentially mutable. While there could be a more fine-grained scheme in the future
that allows mutable references if they aren't "leaked" to the final value, a more
conservative approach was chosen for now. `const fn` do not have this problem, as
the borrow checker will prevent the `const fn` from returning new mutable references.
```